src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -101,10 +100,13 @@ def initialize_jobs(self):
                     for child_id in job_state.get("child_jobs", [])
                 ]
 
-            self.register_new_job(
-                job=Job(job_state, children=child_jobs),
-                refresh=int(job_state.get("status") not in TERMINAL_STATUSES),
-            )
+            job = Job(job_state, children=child_jobs)
+
+            refresh = not job.was_terminal
+            if cell_ids is not None:
+                refresh &= job.in_cells(cell_ids)
```

Comment:
what is this code doing? Should it just be `refresh = job.in_cells(cell_ids)`?

Reply:
`refresh = not job.was_terminal and job.in_cells(cell_ids) if cell_ids else not job.was_terminal`?

Reply:
Looks good. I'd add a comment so that lazy people don't have to work through the logic. 🙂

Reply:
Is the operator correct?

```py
refresh &= job.in_cells(cell_ids)
```

It's simpler to use `=`, and people won't have to go and look up what `&=` means